Source/Core/Core/PowerPC/JitArm64/JitArm64_FloatingPoint.cpp

```diff
@@ -345,78 +345,73 @@ void JitArm64::fcmpx(UGeckoInstruction inst)
 	JITDISABLE(bJITFloatingPointOff);
 
 	u32 a = inst.FA, b = inst.FB;
-	int crf = inst.CRFD;
 
 	ARM64Reg VA = fpr.R(a, REG_IS_LOADED);
 	ARM64Reg VB = fpr.R(b, REG_IS_LOADED);
 
-	ARM64Reg WA = gpr.GetReg();
-	ARM64Reg XA = EncodeRegTo64(WA);
-
-	FixupBranch pNaN1, pNaN2, pNaN3, pLesser, pGreater;
-	FixupBranch continue1, continue2, continue3;
-	ORR(XA, ZR, 32, 0, true);
+	FixupBranch pLesser, pGreater;
 
 	if (a != b)
 	{
 		m_float_emit.FCMP(EncodeRegToDouble(VA), EncodeRegToDouble(VA));
 
-		// if (B != B) or (A != A), goto NaN's jump target
-		pNaN1 = B(CC_NEQ);
+		// if (B != B) or (A != A), Fall back to interpreter
+		if (CC_NEQ)
```

Comment:
No offensive, but I guess you did not understand the idea of a JIT. On the interpreter, this function would be called whenever the CPU executes a fcmp instruction. But on a JIT, we call this code here just once. This code here will not compare two floats, it will generate arm64 assembly which is able to compare floats.
So there are two kinds of flow control: Once on generating this assembly like this one here. This step might optimize based on the instruction itself, but as the compare isn't done right now, it must not do any assumptions about the input values. On the other hand, there is the B() flow control. This is an ARM assembly for branching, so this will happen while executing this ARM instructions. But for this kind of flow control, you must always generate the assembly for all possible ways.